guide/controller-programming.md

```diff
@@ -34,25 +34,21 @@ files. These header files must be included using statements like `#include
 Like with any regular C code it is also possible to include the standard C
 headers, e.g. `#include <stdio.h>`. A call to the initialization function
 `wb_robot_init()` is required before any other C API function call. This
-function initializes the communication between the controller and Webots. Note
-that  `wb_robot_init()` exists only in the C API, it does not have any
-equivalent in the other supported programming languages.
+function initializes the communication between the controller and Webots.
+`wb_robot_cleanup()` does the opposite: it closes the communication between the controller and Webots to terminate the controller smoothly. Note that  `wb_robot_init()` and `wb_robot_cleanup()` exist only in the C API, they do not have any equivalent in the other supported programming languages.
 
 Usually the highest level control code is placed inside a `for` or a `while`
 loop. Within that loop there is a call to the `wb_robot_step()` function. This
 function synchronizes the controller's data with the simulator. The function
 `wb_robot_step()` needs to be present in every controller and it must be called
 at regular intervals, therefore it is usually placed in the main loop as in the
-above example. The value 32 specifies the duration of the control steps, i.e.
+above example. The value 32 specifies the duration of the control steps, i.e.,
 the function  `wb_robot_step()` shall compute 32 milliseconds of simulation and
 then return. This duration specifies an amount of simulated time, not real (wall
-clock) time, so it may actually take 1 millisecond or one minute of CPU time,
+clock) time, so it may actually take 1 millisecond or one minute of real time,
 depending on the complexity of the simulated world.
 
-Note that in this "Hello World!" example the `while` loop has no exit condition,
-hence the return statement is never reached. It is usual to have an infinite
-loop like this in the controller code: the result is that the controller runs as
-long as the simulation runs.
+Note that in this "Hello World!" example, the exit condition of the `while` loop is the return value of the `wb_robot_step()` function. This function will indeed return `-1` when Webots terminates the controller. Therefore, in this example, the control loop will run as long as the simulation runs. When the loop exists, no further communication with Webots is possible and the only option is to confirm to Webots to close the communication by calling `wb_robot_cleanup()`.
```

Comment:
If you prefer like this then it is ok.
But there is no link to the following section and when reading the documentation for the first it is not clear why Webots should terminate the controller. So personally I think it would be better to already give some information here.